### DISCUSSION TITLE
Processing post-creation actions...

### DISCUSSION THREAD

**Discussion by DrJickson** [MonoGame/MonoGame#8528] [General]
Hey all, I'm trying to follow the Getting Started tutorial. I've installed vscode 1.94.2 along with the C# Dev Kit, installed the MonoGame Templates `dotnet new install MonoGame.Templates.CSharp` , and installed MonoGame for VSCode extension.

When i get to the bottom of this page:
https://docs.monogame.net/articles/getting_started/2_choosing_your_ide_vscode.html?tabs=windows

I run the `dotnet new mgdesktopgl` and this is the output I get:
`
PS E:\Coding\Project> dotnet new mgdesktopgl --force
The template "MonoGame Cross-Platform Desktop Application" was created successfully.

Processing post-creation actions...
Restoring E:\Coding\Project\Project.csproj:
  E:\Coding\Project\Project.csproj                                                                    Restore (1144.8s)
`

The problem I have here is that I don't know what it's doing, or why its taking so long. This time around its been running this "Restore" for about 20 minutes. You'll notice I used the "--force" option, because I previously interrupted it almost 40 minutes in. 

Can someone shed some light on this? I've tried google extensively the past hour or so, and the problem is that half the things that come up in the search are how to restore deleted files or such in VSCode. 

I don't have the most powerful computer, but it's still a modern processor that doesn't struggle with most other tasks. Its an AMD Ryzen 7 7735HS (certainly not a gaming rig, though) .

Thanks in advance!

**Comment by mandala-dev** [comment]
I don't have any experience using MonoGame with VS Code, so I can't answer the real question, but just out of curiosity: why not giving it a shot with VS 2022 Community Edition? The MonoGame experience there is quite pleasant. I'm using that one on a much less powerful CPU (i5, 4 cores) and it works very well.

**Comment by Brett208** [comment]
@DrJickson 

I use VS and not VS Code as well, so take this advice with a bit of grain of salt, but will try to give some generic troubleshooting.

Have you checked if there is a way to up the verbosity of the output logging when running `dotnet new mgdesktopgl`? This might give some more granularity on what is happening.

Assuming it is fetching information from the internet (I'm not sure if it is or not), how is your network connection?

Could you verify after taking about 19 minutes to initialize, you are able to compile the new project and show the famous cornflower blue MonoGame window? How long does it take to compile and start running the application? If your compilation time post creation of the project is now short, you could be off to the races (although doesn't answer what happened to the original 19 minutes).

Also, now that you have successfully created your first template, does it still take 19 minutes to create another new project?

Hope some of this is marginally helpful...

-Brett

**Comment by DrJickson** [comment]
Hi @Brett208 and @LemiBijafra 
I have VSCode because it was recommended for a C# Course I am working on - this course is only for the basics. 
I may move to VS Community in the future, but for now I want to finish the course as it has everything outlined. 

I was able to create a new test project and compile it with no issues. I've also taken a few steps through the tutorial and have the ball on the screen and can move it around as expected.

The terminal output still shows 'Determining projects to restore...' the lists some tools, then shows 'Restore was successful' when compiling. But it only takes a few seconds now.
So it seems it only needs to happen once. 

I do have exceptionally slow internet, barely 1Mbps with long latency, over 700ms. So I suspect you're correct about it being related to the network connection.

So I guess the question I still don't know the answer to, what does it mean when its restoring projects?

Thanks again for responding!

- Dr. J

**Reply by Apostolique** [reply]
Since your internet is slow, it takes a while to set up MonoGame initially because the Content Pipeline is something like ~300 MB.

If you want to see my setup for vscode, I write about it here: <https://learn-monogame.github.io/how-to/get-started/> and <https://learn-monogame.github.io/how-to/develop-vscode/>.

**Reply by DrJickson** [reply]
Thanks for this! I'm definitely going to check out those links!

**Comment by MonogameGuy** [comment]
>what does it mean when its restoring projects?

The project is checking for the latest version of MonoGame, and downloading it if needed.

The downloaded files are cached, per-user, and used for subsequent project creations.

The process was not designed for slow or metered connections. Disconnecting from the internet when creating, or building a project could save time.
